### PR TITLE
If applied this commit implements a flexible social login mechanism with github login completely configured & working already

### PR DESCRIPTION
**Pull request by introwit**
**The social login mechanism that has been implemented tends to be completely flexible and easy to understand having the following pros:**

- Routes & controller methods are completely dynamic for any given social login service.
- A middleware to protect & check login attempts against our defined social login services.
- Any social login can be initiated within a single minute of configuration (How that can be done is explained later in this comment)
- Any social login can be revoked within a minute without affecting other logins.
- **Handles all the duplicate logins & signups scenario with breeze**

The Code flow : 

- Just 2 flexible routes to handle all of the social login services
- I have tried to keep the SocialLoginController code as readable as possible & _yes I have reordered the imported namespaces alphabetically as you taught me last time_ :)
- Added a middleware to protect & check login attempts against our defined social login services in the _config/social.php_

**Quick guide**

- For the "GitHub" sign in button the css has been added to a new _github.scss file and imported at the top in app.scss and that has been compiled already with Laravel elixir.
- As expected Laravel Socialite is being used so that should be installed using `composer install` 
_NOTE : SOCIALITE'S V 2.0 SHOULD BE USED WITH OUR 5.3 APP SO  a composer install command is recommended as the composer.json has v2.0 for it._
- Next to migrate the new required social login table run `php artisan migrate`
- Lastly update your .env with GitHub app credentials , for correct keys refer to config/services.php

**To Add a new social login**

- Just add the credentails in .env & update the config/services respectively.
- Update the config/social.php array to keep the middleware check active for it.
- That's it! 

**To Remove any login at any time without affecting other stuff**

- Just remove the respective one from config/social.php
- And you are done!

Sorry for the long read , I hope you don't mind Matt :)

Lastly , here is the screenshort of the github button implemented! Tried hard to make it look as near as possible with @adamwathan 's design in this github login issue #56  

github_ui

Sorry Adam if its not as good as you expected, I am not good with the front-end stuff but still tried my best :)

### CODE REVIEW

**Comment by mattstauffer** [comment]
@introwit Looking forward to looking over this--thanks for your work on it!

**Comment by introwit** [comment]
No problem @mattstauffer I know you have been busy. Also I am still 20 , but I have heard from my parents that being a good parent is really tough and I know you have been busy with the little Jr. Stauffer , I hope the champ is doing good too. Much love to the baby, say my hello :)

File: resources/assets/sass/_github.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#github-login-panel {
+	text-align: center;
+}
+
+#github-login-panel .btn {
+	margin-bottom: 0px;
```
Comment (imjohnbon) [inline resources/assets/sass/_github.scss]: A few notes about CSS styling for this file:

1. CSS properties should be alphabetized.
2. Properties should also be indented by 4 spaces.
3. There should be a newline after each set.

File: app/Http/Kernel.php [hunk 2]
```diff
@@ -61,5 +61,7 @@ class Kernel extends HttpKernel
         'oauth-user' => \LucaDegasperi\OAuth2Server\Middleware\OAuthUserOwnerMiddleware::class,
         'oauth-client' => \LucaDegasperi\OAuth2Server\Middleware\OAuthClientOwnerMiddleware::class,
         'check-authorization-params' => \LucaDegasperi\OAuth2Server\Middleware\CheckAuthCodeRequestMiddleware::class,
+
```
Comment (imjohnbon) [inline app/Http/Kernel.php]: We can get rid of the extra newline here.

**Comment by imjohnbon** [comment]
@introwit Hey, great work on this! Tested out locally and it seems to function great. Requested a few minor changes that should help make this ready to go.

**Comment by introwit** [comment]
@imjohnbon done. Sorry for the dumb spacing mistakes in the CSS. Also learned the ordering of properties should be alphabetised,  thanks John :) And your profile bio is :heart:

**Comment by imjohnbon** [comment]
@introwit Thanks for making the cleanups! This is looking really good. Just a few more notes:

1. Right now, the login form has the "Login with Github" button and then a "Don't have a github account?" link that links off to a standalone login page. Given that to start most people will have non-github accounts, we'd like to bring back the login form to the login box and give them both options at the same time. So something like this (just a quick mock-up I made):

screen shot 2017-06-20 at 11 11 42 am

2. As I was doing some final testing, I also realized the Github Login button overlaps a bit at certain screen sizes (mainly tablet as far as I can tell). Here's a screenshot:

screen shot 2017-06-20 at 10 58 05 am

It probably just needs some padding adjustments or something like that.

Thanks again for all your hard work on this! Let me know if you have any questions. Also, thanks for the love for my description! 😁

**Comment by introwit** [comment]
@imjohnbon I am working on a fresh PR for this entire contribution because have learned a lot about working with CSS after the day I made this PR in Jan to till now. Also I am extremely sorry for this delayed response I don't know how I missed this, so sorry John :(

**Comment by imjohnbon** [comment]
@introwit No need to apologize! We appreciate all your hard work on this PR, it's really close to being perfect. If you need help finishing it out at any point just let us know. Otherwise great work so far!

**Comment by imjohnbon** [comment]
@introwit Hey! Just responding to your tweet here since it's a little bit hard to explain over twitter. :)

The majority of Symposium's custom SCSS lives in [resources/assets/sass/app.scss](https://github.com/tightenco/symposium/tree/master/resources/assets) right now. So that's where you'd be putting any new SCSS as well. You can then compile the SCSS to CSS using Gulp by running "gulp" on the command line or "gulp watch" to compile changes as you make them. Here's a link to an [old Laracasts video](https://laracasts.com/lessons/gulp-this) that details setting up/using Gulp if you're unfamiliar with it. Hope this helps! Let me know if you have any further questions.

**Pull request closed without merging**